asllib/carpenter/README.md

```diff
@@ -0,0 +1,58 @@
+# ASL Carpenter
+
+Carpenter is an AST generator for ASL, working in two different modes:
+
+1. Enumeration of all ASTs by increasing size;
+2. Random generation of ASTs following type-checking constraints.
+
+Both modes support constraints on syntax nodes available, and on execution or
+type-checking rules used by ASLRef to execute it.
+
+It can also serve as a fuzzing tool, or a mass execution tool.
+
+## Installation
+
+Dependencies:
+```bash
+opam install dune qcheck-core zarith re feat fix logs
```

Comment:
I followed the instructions and got this error:
```
File "asllib/carpenter/test/dune", line 5, characters 32-38:
5 |   (libraries carpenter_lib feat qcheck asllib))
                                    ^^^^^^
Error: Library "qcheck" not found.
```
I replaced `qcheck-core` by `qcheck` and ran `dune build asllib/carpenter` again and got this error:
```
File "asllib/carpenter/test/dune", line 5, characters 27-31:
5 |   (libraries carpenter_lib feat qcheck asllib))
                               ^^^^
Error: Library "feat" not found.
```
So I installed `feat`b resulting in the following `opam` command:
```
```suggestion
opam install dune qcheck zarith re feat fix logs feat
```

However compilation of carpenter still fails:
```
File "asllib/carpenter/dune", line 14, characters 35-43:
14 |  (libraries cmdliner carpenter_lib logs.cli))
                                        ^^^^^^^^
Error: Library "logs.cli" in /home/maranget/.opam/5.1.1/lib/logs is hidden
(unsatisfied 'exist_if').
```
Something may be wrong with my installation. In any case guidance would be appreciated.

```

Reply:
At some point cmdliner is also missing from the list of dependencies.